pkg/tfgen/docs_test.go

```diff
@@ -898,6 +897,110 @@ throw new Exception("!");
 	assert.NotContains(t, string(schemaBytes), "{{% //examples %}}")
 }
 
+func TestParseTFMarkdown(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		info                    tfbridge.ResourceOrDataSourceInfo
+		providerInfo            tfbridge.ProviderInfo
+		kind                    DocKind
+		resourcePrefix, rawName string
+
+		docRules *tfbridge.DocRuleInfo
+
+		fileName     string
+		fileContents []byte
+
+		expected entityDocs
+	}
+
+	// A pre-configured test case
+	tc := func(configure func(tc *testCase)) testCase {
+		tc := testCase{
+			kind:           ResourceDocs,
+			resourcePrefix: "pkg_",
+			rawName:        "pkg_mod1_res1",
+
+			fileName: "mod1_res1.md",
+			expected: entityDocs{
+				Arguments:  map[string]*argumentDocs{},
+				Attributes: map[string]string{},
+			},
+		}
+		configure(&tc)
+		return tc
+	}
+	// Assert that file contents match the expected description.
+	desc := func(fileContents, expected string) testCase {
+		return tc(func(c *testCase) {
+			c.fileContents = []byte(fileContents)
+			c.expected.Description = expected
+		})
+	}
+
+	tests := []testCase{
+		desc(`
```

Comment:
This is good consider adding some whitespace and indent for readability.. since funcs an not records are used and without indent it's hard to quickly see which arg is which :)